cpp/src/arrow/util/parsing.h

```diff
@@ -419,8 +419,9 @@ class StringConverter<TimestampType> {
         *out = std::chrono::duration_cast<std::chrono::nanoseconds>(duration).count();
         return true;
     }
-    // Unreachable
+    // Unreachable, but suppress compiler warning
     assert(0);
+    *out = 0;
```

Comment:
I'd set this in the `default` case, this is a bit friendlier to callers if they don't check the result.

Reply:
Hmm... what do you mean? It should do the same in the end.

Reply:
Maybe it's just an old habit of using `-Wswitch-enum`.